### PR TITLE
Fix selenium tests in factory package

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### What does this PR do?
Fixes tests in `factory` package

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10919

### CODE REVIEW

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 1]
```diff
@@ -462,6 +465,18 @@ private void checkLanguageServerInitialized() {
     }
   }
 
+  private CompletableFuture<ServerCapabilities> getOrInitializeLanguageServer() {
+    if (serverCapabilities.get() == null) {
```
Comment (tsmaeder) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: It makes no sense to do an AtomicReference and synchronized protection.

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 2]
```diff
@@ -462,6 +465,18 @@ private void checkLanguageServerInitialized() {
     }
   }
 
+  private CompletableFuture<ServerCapabilities> getOrInitializeLanguageServer() {
```
Comment (tsmaeder) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: We should not do this. Whether we decide to initialize a server or not should be done by the server initializer. Don't cache the result, it's not worth it.

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java [hunk 3]
```diff
@@ -11,43 +11,25 @@
...
 @Singleton
 public class MavenProjectInitHandler implements ProjectInitHandler {
 
-  private final WorkspaceSynchronizer workspaceSynchronizer;
-
   @Inject
-  public MavenProjectInitHandler(WorkspaceSynchronizer workspaceSynchronizer) {
-    this.workspaceSynchronizer = workspaceSynchronizer;
-  }
+  public MavenProjectInitHandler(WorkspaceSynchronizer workspaceSynchronizer) {}
```
Comment (tsmaeder) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java]: If we're not using it, remove the parameter.

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java [hunk 4]
```diff
@@ -11,43 +11,25 @@
...
   @Override
-  public void onProjectInitialized(String projectFolder)
-      throws ServerException, ForbiddenException, ConflictException, NotFoundException {
-
-    UpdateWorkspaceParameters params = new UpdateWorkspaceParameters();
-    params.setAddedProjectsUri(Collections.singletonList(prefixURI(projectFolder)));
-
-    workspaceSynchronizer.syncronizerWorkspaceAsync(params);
-  }
+  public void onProjectInitialized(String projectFolder) {}
```
Comment (tsmaeder) [inline plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java:33]: This looks like we're not using this project initializer anymore. Can you explain how this made a selenium test fail and why we need to change it?
Generally, I think this kind of "meaningful" change would warrant a separate bug report and PR.

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java [hunk 5]
```diff
@@ -221,7 +221,8 @@ public ProjectConfigDto updateProject(String wsPath, ProjectConfigDto projectCon
       throws NotFoundException, ConflictException, ForbiddenException, ServerException, IOException,
           BadRequestException {
     if (wsPath != null) {
-      projectConfigDto.setPath(absolutize(wsPath));
+      wsPath = absolutize(wsPath);
```
Comment (tsmaeder) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java:224]: Again, what was the bug here?

**Review by tsmaeder — CHANGES_REQUESTED** [review]

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java (hunk 5 above)
Comment (tolusha) [reply wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java:224]: `absolutize` ensure that project path starts with `/`. 
Without that it leads it emitting `ProjectCreatedEvent` for already registered projects.

Have a look at code below:
```
    boolean registeredEarly = projectManager.isRegistered(wsPath);
    RegisteredProject updated = projectManager.update(projectConfigDto);
```
Without `absolutize` registration checking and updating will be performed for different project paths.

File: plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java (hunk 4 above)
Comment (tolusha) [reply plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/projecttype/handler/MavenProjectInitHandler.java:33]: Using `MavenProjectInitHandler` leads us to endless cycle:
1. Project with problems is imported
2. `JavaLanguageServerExtensionService.updateWorkspace` is invoked
3. `JavaLanguageServerExtensionService.updateProjectsWithProblems` is invoked
4. `MavenProjectInitHandler.onProjectInitialized` is invoked.
5. `WorkspaceSynchronizer.syncronizerWorkspaceAsync` is invoked
6. -> 2

**Review by svor — APPROVED** [review]
One point here. All existed factories which use java won't work after merging jdt.ls into master because they don't have `org.eclipse.che.ls.java` installer. It can be a problem.

**Comment by tolusha** [comment]
@tsmaeder Could you have a look again?

**Review by tsmaeder — APPROVED** [review]

**Pull request merged**